server/game/core/card/propertyMixins/UnitProperties.ts

```diff
@@ -101,9 +126,13 @@ export function WithUnitProperties<TBaseClass extends InPlayCardConstructor>(Bas
             let triggeredAbilities = super.getTriggeredAbilities();
 
             // add any temporarily registered attack abilities from keywords
-            if (this._attackKeywordAbilities !== null || this.isBlank()) {
+            if (this._attackKeywordAbilities !== null || this.isBlank()) { //TODO: Why is this isBlank check even here?
                 triggeredAbilities = triggeredAbilities.concat(this._attackKeywordAbilities.filter((ability) => ability instanceof TriggeredAbility));
             }
+            if (this._whenPlayedKeywordAbilities !== null || this.isBlank()) {
+                // TODO: does it even make sense for there to be non-triggered when played keyword abilities? I think not. Maybe Smuggle?
```

Comment:
I think that's a reasonable assumption but if so you should remove the `removeEffectFromEngine` call on line 216 since that's only relevant for constant abilities.